ols/user_data_collection/data_collector.py

```diff
@@ -0,0 +1,245 @@
+"""Collect insights and upload it to the Ingress service.
+
+It collects the data on startup and then it waits for the specified interval.
+"""
+
+import base64
+import io
+import json
+import logging
+import os
+import pathlib
+import subprocess
+import sys
+import tarfile
+import time
+
+import requests
+
+OLS_USER_DATA_PATH = os.environ.get("OLS_USER_DATA_PATH", "./user-data/")
+
+OLS_USER_DATA_COLLECTION_INTERVAL = int(
+    os.environ.get("UDC_COLLECTION_INTERVAL", 2 * 60 * 60)
+)  # 2 hours in seconds
+
+# available ingress urls are:
+# - https://console.redhat.com/ - production - works with refresh token
+# - https://console.stage.redhat.com/ - staging/dev - doesn't work with refresh token
+INGRESS_URL = os.environ.get("INGRESS_URL", "https://console.redhat.com/")
```

Comment:
this one is ok to leave as an env var since there should be no reason to expect us to use any other value except for testing purposes.